test/support/utils.rb

```diff
@@ -36,6 +36,18 @@ def debug_code(program, **options, &block)
       check_line_num!(program)
     end
 
+    # use this to start a debug session with the test program
+    def start_debugger_with(program)
+      write_temp_file(strip_line_num(program))
+      inject_lib_to_load_path
+      pid = spawn("#{RUBY} -r debug/open #{temp_file_path}")
+      sleep(0.5) # wait for the debug server to start
```

Comment:
I would like to know how to decide seconds.

Reply:
I guess you can say it's arbitrary 😛
because there's no way to communicate about the server state atm, the connection will fail right away when the server is not up yet. given that this is just an internal helper to help us write tests, I think it's not worth changing the library's behavior for it.

or perhaps @ko1 has a plan on adding retry & timeout here? https://github.com/ruby/debug/blob/2eed8dfe57cfc8b752001f10515bec0f2cff73e9/lib/debug/client.rb#L75